src/core.c/Rakudo/Internals.pm6

```diff
@@ -1384,15 +1384,15 @@ my class Rakudo::Internals {
             nqp::while(
               nqp::chars(my str $entry = self!next),
               nqp::stmts(
+                (next if $entry eq '.' or $entry eq '..'),
```

Comment:
This does probably need to be removed in the final version as both `$!file` and `$!dir` are configurable (with `$!file` defaulting to matching everything).